### PR TITLE
Plasma cutter EMP interaction.

### PR DESCRIPTION
**Pull request by AnCopper**
## About The Pull Request

Adds an EMP interaction with the plasma cutter which sets the user on fire.

## Why It's Good For The Game
More EMP interactions, why not?

https://github.com/tgstation/tgstation/pull/64073
## Testing Photographs and Procedure

<details>

https://user-images.githubusercontent.com/82319489/150795352-527244a3-b0fe-4c8a-af78-e2de71ce4e3c.mp4

</details>

## Changelog
:cl: SkeletalElite, AnCopper
add: Charged plasma cutters will set their user on fire when hit by an EMP.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
~~This would be made redundant by https://github.com/BeeStation/BeeStation-Hornet/pull/6135~~ Nevermind it sets the user on fire too, but should call parent.

**Comment by AnCopper** [comment]
> ~This would be made redundant by #6135~ Nevermind it sets the user on fire too, but should call parent.

I don't understand what you're trying to tell me.

**Comment by KubeRoot** [comment]
> I don't understand what you're trying to tell me.

Here's a sample from code for what seems to be the proper way to implement `emp_act` overrides:

```dm
/obj/machinery/emp_act(severity)
	. = ..()
	if(use_power && !stat && !(. & EMP_PROTECT_SELF))
		use_power(7500/severity)
		new /obj/effect/temp_visual/emp(loc)
```

1. The base implementation of `emp_act` calls a signal, you need to preserve this behavior.
2. `emp_act` returns a bitflag for EMP resistance, here you care for `EMP_PROTECT_SELF` which would mean the plasma cutter somehow has immunity to EMPs. No idea if this can ever actually happen, but better to code it robustly.

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -177,6 +177,18 @@
 	else
 		..()
 
+/obj/item/gun/energy/plasmacutter/emp_act(severity)
```
Comment (Archanial) [inline code/modules/projectiles/guns/energy/special.dm:180]: missing parent call

Comment (AnCopper) [reply code/modules/projectiles/guns/energy/special.dm:180]: It works tho.

Comment (PowerfulBacon) [reply code/modules/projectiles/guns/energy/special.dm:180]: Read KubeRoot's comment

**Pull request closed without merging**